### PR TITLE
SCIPROD-1859, SCIPROD-1860: [CLIEXPRESS] bug fixes

### PR DESCRIPTION
**Pull request by gabriela-reznik**

### CODE REVIEW

File: src/python/dxpy/cli/dataset_utilities.py [hunk 1]
```diff
@@ -1221,7 +1221,7 @@ def extract_assay_expression(args):
                 )
             )
     elif not args.assay_name:
-        if len(dataset.assays_info_dict["molecular_expression"]) == 0:
+        if dataset.assays_info_dict.get("molecular_expression") is None or len(dataset.assays_info_dict.get("molecular_expression")) == 0:
```
Comment (jkotrs) [inline src/python/dxpy/cli/dataset_utilities.py:1224]: Minor enh.:
```
dataset.assays_info_dict.get("molecular_expression") in [None, 0]:
```

**Review by jkotrs — APPROVED** [review]

File: src/python/dxpy/cli/dataset_utilities.py (hunk 1 above)
Comment (gabriela-reznik) [reply src/python/dxpy/cli/dataset_utilities.py:1224]: Would that really work? One is testing existence but the other length == 0

Comment (jkotrs) [reply src/python/dxpy/cli/dataset_utilities.py:1224]: @gabriela-reznik Oh, my bad! I've missed the `len` completely. You are right, that would work. Pls disregard this comment.:)

**Review by vkuban-dx — APPROVED** [review]

**Review by jkotrs — APPROVED** [review]
Ha, good catch!

**Review by ashkan0x — APPROVED** [review]
Looks great! Thanks for the fixes.

**Pull request merged**